ui/index.js

```diff
@@ -25,4 +29,12 @@ if ($('.student-searchbar').length > 0) {
 }
 
 // Routing
-route();
+// Some pages are server-rendered and have a different structure
+// other than #main so we ignore those.  Other pages add in class names
+// to the body tag that `legacyRouteHandler` works with.  Newer pages
+// should handle routing with react-router inside the `App` component.
+const mainEl = document.getElementById('main');
+if (mainEl) {
```

Comment:
Is `index.js` going to run on every single page/page load in the app?

Reply:
Yep, it already does except for a few admin pages that don't load any JS at all.

Ideally we'd run the routing, then run any other code based on that.  But we're still doing a lot of JS execution in the different legacy.js files as part of the migration from Sprockets.